### ISSUE TITLE
Baseband fails if DADA header payload size doesn't match actual file payload size

### ISSUE THREAD

**Issue by quantumfx** [mhvk/baseband#149]
This happens often for the last file of an observation run. I am working on this issue.

Just to practice documenting, here's the traceback on a test DADA file where this happens:

```
In [61]: b = dada.open('b.dada', 'rs', squeeze=False)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-61-737657ac9e72> in <module>()
----> 1 b = dada.open('b.dada', 'rs', squeeze=False)

~/src/baseband/baseband/dada/base.py in open(name, mode, subset, header, **kwargs)
    482             kwargs['subset'] = subset
    483
--> 484     return opener(name, mode, **kwargs)
    485
    486

~/src/baseband/baseband/vlbi_base/base.py in open(name, mode, **kwargs)
    521                 except Exception:  # pragma: no cover
    522                     pass
--> 523             raise exc
    524
    525     open.__doc__ = (default_open_doc.replace('baseband', fmt) + doc

~/src/baseband/baseband/vlbi_base/base.py in open(name, mode, **kwargs)
    514                              .format(fmt))
    515         try:
--> 516             return classes[cls_type](name, **kwargs)
    517         except Exception as exc:
    518             if not got_fh:

~/src/baseband/baseband/dada/base.py in __init__(self, fh_raw, subset, squeeze)
    236         header = DADAHeader.fromfile(fh_raw)
    237         super(DADAStreamReader, self).__init__(fh_raw, header, subset, squeeze)
--> 238         self._get_frame(0)
    239
    240     @lazyproperty

~/src/baseband/baseband/dada/base.py in _get_frame(self, frame_nr)
    301     def _get_frame(self, frame_nr):
    302         self.fh_raw.seek(frame_nr * self.header0.framesize)
--> 303         self._frame = self.read_frame(memmap=True)
    304         self._frame_nr = frame_nr
    305         assert (self._frame.header['OBS_OFFSET'] ==

~/src/baseband/baseband/dada/base.py in read_frame(self, memmap)
    126             parts of interest by slicing the frame.
    127         """
--> 128         return DADAFrame.fromfile(self.fh_raw, memmap=memmap)
    129
    130

~/src/baseband/baseband/dada/frame.py in fromfile(cls, fh, memmap, valid, verify)
     74         """
     75         header = cls._header_class.fromfile(fh, verify)
---> 76         payload = cls._payload_class.fromfile(fh, header=header, memmap=memmap)
     77         return cls(header, payload, valid=valid, verify=verify)
     78

~/src/baseband/baseband/dada/payload.py in fromfile(cls, fh, header, memmap, payloadsize, **kwargs)
     94             words = np.memmap(fh, mode=mode, dtype=cls._dtype_word,
     95                               offset=offset, shape=None if payloadsize is None
---> 96                               else (payloadsize // cls._dtype_word.itemsize,))
     97             fh.seek(offset + words.size * words.dtype.itemsize)
     98         return cls(words, header=header, **kwargs)

/usr/local/lib/python3.6/site-packages/numpy/core/memmap.py in __new__(subtype, filename, dtype, mode, offset, shape, order)
    262         bytes -= start
    263         array_offset = offset - start
--> 264         mm = mmap.mmap(fid.fileno(), bytes, access=acc, offset=start)
    265
    266         self = ndarray.__new__(subtype, shape, dtype=descr, buffer=mm,

ValueError: mmap length is greater than file size
```

**Comment by cczhu** [comment]
DADA's currently designed so that more than one frame can be read from a single file - this is useful when a [sequentialfile](https://baseband.readthedocs.io/en/latest/baseband/helpers/index.html) handle is passed.  If the reader sets the payloadsize to be the entire file, that would be incorrect.  Seeking for a header in files as large as DADA ones is also impractical.

One possibility is to let the user override the payloadsize, then let them deal with it if it's wrong.  I don't know how I feel about that.  Also, have we ever encountered a sequence of DADA files that are different sizes?  @mhvk, what are your thoughts?

**Comment by mhvk** [comment]
@quantumfx  and I are currently thinking of simply relaxing the check on the payloadsize for the last frame - this is uniquely defined.

**Comment by cczhu** [comment]
Solved by #228
